sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -267,12 +267,32 @@ private[sql] abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
   }
 
   object CartesianProduct extends Strategy {
+    def createCartesianProduct(left: LogicalPlan, right: LogicalPlan): SparkPlan = {
```

Comment:
instead of creating a new function, i think it would be clearer if you just inline the CanBroadcast matching into the rule itself.